File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int mxSum = nums[0];
        int currSum = nums[0];
        
        for(int i = 1; i < nums.length; i++)
        {
            currSum = Math.max(nums[i],nums[i]+currSum);
            
            mxSum = Math.max(currSum, mxSum);
        }
        
        return mxSum;
    }
}

File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int currSum=nums[0],sum = nums[0];
        for(int i = 1; i < nums.size(); i++)
        {
            currSum = max(nums[i]+currSum, nums[i]);
            sum = max(sum,currSum);
        }
        return sum;
    }
};